### PR TITLE
Get manufacturer serial number

### PR DESCRIPTION
**Pull request by M1chaelTran**
Got my solution from: https://stackoverflow.com/questions/14161282/serial-number-from-samsung-device-running-android

### CODE REVIEW

File: android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java [hunk 1]
```diff
@@ -93,6 +94,26 @@ private Boolean is24Hour() {
     return android.text.format.DateFormat.is24HourFormat(this.reactContext.getApplicationContext());
   }
 
+  private String getSerialNumber(String className){
+    String serial = "unknown";
```
Comment (machour) [inline android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: use `null` instead

File: android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java [hunk 2]
```diff
@@ -93,6 +94,26 @@ private Boolean is24Hour() {
...
+    try {
+      Class<?> c = Class.forName("android.os.SystemProperties");
+      Method get = c.getMethod("get", String.class, String.class);
+      serial = (String) get.invoke(c, className, "unknown");
+    } catch (Exception ignored) {}
+    return serial;
+  }
+
+  private String getManufacturerSerialNumber() {
+    String serial = "unknown";
```
Comment (machour) [inline android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: use `null` instead

File: android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java [hunk 3]
```diff
@@ -93,6 +94,26 @@ private Boolean is24Hour() {
...
+    try {
+      serial = this.getSerialNumber("ril.serialnumber");
+      if(serial.equals("unknown"))
```
Comment (machour) [inline android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: curly brackets please {}

File: README.md [hunk 4]
```diff
@@ -182,43 +182,44 @@ var DeviceInfo = require('react-native-device-info');
...
+| [getTotalMemory()](#gettotalmemory)                            | `number`            |  ✅  |   ✅   |   ❌   | 0.14.0 |
+| [getUniqueID()](#getuniqueid)                                  | `string`            |  ✅  |   ✅   |   ✅   | ?      |
+| [getUserAgent()](#getuseragent)                                | `string`            |  ✅  |   ✅   |   ❌   | 0.7.0  |
+| [getVersion()](#getversion)                                    | `string`            |  ✅  |   ✅   |   ✅   | ?      |
+| [is24Hour()](#is24hour)                                        | `boolean`           |  ✅  |   ✅   |   ✅   | 0.13.0 |
+| [isEmulator()](#isemulator)                                    | `boolean`           |  ✅  |   ✅   |   ✅   | ?      |
+| [isPinOrFingerprintSet()](#ispinorfingerprintset)              | (callback)`boolean` |  ✅  |   ✅   |   ❌   | 0.10.1 |
+| [isTablet()](#istablet)                                        | `boolean`           |  ✅  |   ✅   |   ✅   | ?      |
+| [getManufacturerSerialNumber()](#getManufacturerSerialNumber)  | `string`            |  ❌  |   ✅   |   ❌   | 0.15.2 |
```
Comment (machour) [inline README.md]: the link part should be lowercased.
additionally, could you put that line right under `getManufacturer()` so that we preserve sorting?

Comment (machour) [reply README.md]: put `next` as version information

File: README.md [hunk 5]
```diff
@@ -773,6 +774,19 @@ Tells if the device is a tablet.
 const isTablet = DeviceInfo.isTablet(); // true
 ```
 
+---
+
+### getManufacturerSerialNumber()
+
+Returns device manufacturer sercial number. Tested on Samsung devices.
```
Comment (machour) [inline README.md]: Typo in "serial"

**Comment by jawu-tesla** [comment]
@M1chaelTran  @machour, 

is there any update status of this PR? when will this be merged? I have an immediate need to use the manufacturing serial number and want to know if this is going to be merged anytime soon or if I should just fork this repo and make the changes myself

**Pull request closed without merging**